sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -152,8 +168,9 @@ case class HashAggregateExec(
   override def usedInputs: AttributeSet = inputSet
 
   override def supportCodegen: Boolean = {
-    // ImperativeAggregate is not supported right now
-    !aggregateExpressions.exists(_.aggregateFunction.isInstanceOf[ImperativeAggregate])
+    // ImperativeAggregate and filter predicate are not supported right now
+    !(aggregateExpressions.exists(_.aggregateFunction.isInstanceOf[ImperativeAggregate]) ||
+        aggregateExpressions.exists(_.filter.isDefined))
```

Comment:
We cannot support this in the codegen mode? Technically hard? If we support this filter expr in agg, I personally think we should support in the codegen mode first.

Reply:
Thanks for your review. I think this is a bigger change and I want create two PR to make things simple.
Another reason is I took a lot of time in other jobs.